daft/utils.py

```diff
@@ -10,6 +10,20 @@
 ARROW_VERSION = tuple(int(s) for s in pa.__version__.split(".") if s.isnumeric())
 
 
+def in_notebook():
+    """Check if we are in a Jupyter notebook."""
+    try:
+        from IPython import get_ipython
+
+        if "IPKernelApp" not in get_ipython().config:  # pragma: no cover
```

Comment:
Would this also return true when in a ipython shell?

this is what tqdm does:
https://github.com/tqdm/tqdm/blob/master/tqdm/autonotebook.py